presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -106,7 +106,7 @@ public boolean contains(Block block, int position)
         }
     }
 
-    public void add(Block block, int position)
+    public boolean add(Block block, int position)
```

Comment:
Would this refactor suffice to avoid adding a separate addNonNull method.
```
      if (block.mayHaveNull()) {
            // containsNullElement flag is maintained so contains() method can have shortcut for null value
            if (block.isNull(position)) {
                containsNullElement = true;
            }
        }
```

Reply:
i believe I added a comment here but I don't see it! Anyway, the idea is that we are adding a non-null. It's not that the set doesn't contain null. So I want to keep it to the point.

Reply:
Had an offline chat with @kaikalur . LGTM